workloads/virtual-machines/disks-and-volumes.md

```diff
@@ -373,3 +375,98 @@ spec:
 
 Ephemeral VMs very often come with read-only root images and limited tmpfs space. In many cases this is not enough to install application dependencies and provide enough disk space for the application data. While this data is not critical and thus can be lost, it is still needed for the application to function properly during its lifetime. This is where an `emptyDisk` can be useful. An emptyDisk is often used and mounted somewhere in `/var/lib` or `/var/run`.
 
+### configMap
+
+A `configMap` is a reference to a [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) in Kubernetes. An extra `iso` disk will be allocated which has to be mounted on a VM. To mount the `configMap` users can use a `cloudInit` and disk serial number. The `configMapName` needs to be specified for a reference to the created kubernetes `ConfigMap` with the same name.
+
+Example: Attach the `configMap` to a VM and use a `cloudInit` to mount the `iso` disk:
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VirtualMachineInstance
+metadata:
+  name: testvmi-configmap
+spec:
+  terminationGracePeriodSeconds: 5
+  domain:
+    resources:
+      requests:
+        memory: 64M
+    devices:
+      disks:
+      - name: registrydisk
+        volumeName: registryvolume
+        disk:
+          bus: virtio
+      - name: configmap-disk
+        volumeName: configmap-volume
+        disk: {}
+      - name: cloudinit-disk
+        volumeName: cloudinit-volume
+        disk:
+          bus: virtio
+        # serial has to be set
+        serial: CVLY623300HK240D
+  volumes:
+    - name: registryvolume
+      registryDisk:
+        image: kubevirt/cirros-registry-disk-demo:latest
+    - name: configmap-volume
+      configMap:
+        configMapName: app-config
+    - name: cloudinit-volume
+      cloudInitNoCloud:
+        userData: |
+          #!/bin/sh
+
+          # use a disk serial number to mount the ConfigMap
+          "sudo mount /dev/$(lsblk --nodeps -no name,serial | grep CVLY623300HK240D | cut -f1 -d' ') /blub"
```

Comment:
Did you try the example? I think you need  to at least also have a `mkdir /blub`.

Reply:
Real examples are added. I tested them.